### PR TITLE
Provide API for deriving reflog of HEAD

### PR DESCRIPTION
**Pull request by PawelLipski**
Will be useful for selecting the most recently checked out branches for layout discovery.

### CODE REVIEW

File: gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreHeadSnapshot.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.virtuslab.gitcore.api;
+
+import io.vavr.collection.List;
+import io.vavr.control.Option;
+
+/**
+ * An immutable snapshot of git repository's HEAD for some specific moment in time.
+ * Each {@code get...} method is guaranteed to return the same value each time it's called on a given object.
+ */
+public interface IGitCoreHeadSnapshot {
```
Comment (PawelLipski) [inline gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreHeadSnapshot.java:10]: image

Comment (mkondratek) [reply gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreHeadSnapshot.java:10]: This red logo text and the strike outgoing from bottom left to the upper right reminds me of something :rofl: :flight_departure:

Comment (mkondratek) [reply gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreHeadSnapshot.java:10]: This red logo text and the strike outgoing from bottom left to the upper right reminds me of something :rofl: :flight_departure:

Comment (PawelLipski) [reply gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreHeadSnapshot.java:10]: Wow it's actually an **Indonesian** movie

image

Comment (PawelLipski) [reply gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreHeadSnapshot.java:10]: And fun fact, high in IMDb results there's also 2011 Thai _Headshot_ xD

image

**Pull request merged**